File: homework1/1356_Sort_Integers_by_The_Number_of_1_Bits.py
from typing import List
class Solution:
    # nếu muốn cả 2 đều giảm thì thêm dấu trừ phias trước
    # Vd - self.countBit1(x)
    # vd -x
    def sortByBits(self, arr: List[int]) -> List[int]:
        sorted(arr, key= lambda x :(
            self.countBit1(x), x
        ))

    def countBit1(a):
        count  = 0
        while a > 0:
            if a % 2 == 1: count += 1
            a = a / 2
        return count

File: homework1/1356_Sort_Integers_by_The_Number_of_1_Bits.cpp
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;
class Solution {
public:
    //  this là để cho phép truy câp hàm hoặc biến ở trong class
    vector<int> sortByBits(vector<int>& arr) {
        sort(arr.begin(), arr.end(), [this](int a, int b){
            int ca = countBit1(a);
            int cb = countBit1(b);
            if ( ca != cb) return ca < cb;
            return a < b;
        });
        return arr;
    }

    int countBit1(int a){
        int count = 0;
        while(a > 0){
            if ( a % 2 == 1) count++;
            a = a/2;
        }
        return count;
    }
};

int main(){
    Solution sol;
    cout << sol.countBit1(8) << endl;
    return 0;
}

